### ISSUE TITLE
Exceptions caused by unbound `this` in ref callbacks

### ISSUE THREAD

**Issue by gyng** [uber/react-map-gl#336]
react-map-gl 3.0.1

To reproduce:

1. `npm start`
2. View Custom Overlay example

```
@ canvas-overlay.js:65

Uncaught TypeError: Cannot set property '_canvas' of undefined
    at _canvasLoaded (canvas-overlay.js:65)
    at attachRef (ReactRef.js:20)
    at Object.ReactRef.attachRefs (ReactRef.js:42)
    at ReactDOMComponent.attachRefs (ReactReconciler.js:23)
    at CallbackQueue.notifyAll (CallbackQueue.js:76)
    at ReactReconcileTransaction.close (ReactReconcileTransaction.js:80)
    at ReactReconcileTransaction.closeAll (Transaction.js:209)
    at ReactReconcileTransaction.perform (Transaction.js:156)
    at ReactUpdatesFlushTransaction.perform (Transaction.js:143)
    at ReactUpdatesFlushTransaction.perform (ReactUpdates.js:89)
```

Caused by https://github.com/uber/react-map-gl/commit/b2ce2dfc45e4559e879eef897518abb1fc068f64?

Changing `ref: this._canvasLoaded,` to `ref: this._canvasLoaded.bind(this)` in `canvas-overlay.js` fixes it for `CanvasOverlay`.

**Comment by abmai** [comment]
@gyng Thanks for flagging this! This should be fixed in `v3.0.3` - can you double check?

**Comment by gyng** [comment]
@abmai Thanks, it looks fixed! I'll go ahead and close the issue.
